sos/report/plugins/nvme.py

```diff
@@ -26,24 +26,44 @@ class Nvme(Plugin, IndependentPlugin):
     kernel_mods = ('nvme', 'nvme_core')
 
     def setup(self):
+        _subdir1 = "nvme_smartctl"
+        _subdir2 = "nvme_fw_smart_error_logs"
+        _subdir3 = "nvme_list-ns"
+        _subdir4 = "nvme_list-ctrl"
+        _subdir5 = "nvme_id-ctrl_-H"
+        _subdir6 = "nvme_id-ns_-H"
+        _subdir7 = "nvme_show-regs"
+
         self.add_copy_spec("/etc/nvme/*")
         self.add_cmd_output([
             "nvme list",
             "nvme list-subsys",
         ])
 
-        cmds = [
+        smartctl_cmds = [
             "smartctl --all %(dev)s",
-            "smartctl --all %(dev)s -j",
-            "nvme list-ns %(dev)s",
+            "smartctl --all %(dev)s -j"
+        ]
+
+        nvme_logs_cmds = [
             "nvme fw-log %(dev)s",
-            "nvme list-ctrl %(dev)s",
-            "nvme id-ctrl -H %(dev)s",
-            "nvme id-ns -H %(dev)s",
             "nvme smart-log %(dev)s",
-            "nvme error-log %(dev)s",
-            "nvme show-regs %(dev)s"
+            "nvme error-log %(dev)s"
         ]
-        self.add_device_cmd(cmds, devices='block', whitelist='nvme.*')
+
+        self.add_device_cmd(smartctl_cmds, devices='block', whitelist='nvme.*',
+                            subdir=_subdir1)
+        self.add_device_cmd(nvme_logs_cmds, devices='block',
+                            whitelist='nvme.*', subdir=_subdir2)
+        self.add_device_cmd("nvme list-ns %(dev)s", whitelist='nvme.*',
+                            devices='block', subdir=_subdir3)
+        self.add_device_cmd("nvme list-ctrl %(dev)s", whitelist='nvme.*',
+                            devices='block', subdir=_subdir4)
+        self.add_device_cmd("nvme id-ctrl -H %(dev)s", whitelist='nvme.*',
+                            devices='block', subdir=_subdir5)
+        self.add_device_cmd("nvme id-ns -H %(dev)s", whitelist='nvme.*',
+                            devices='block', subdir=_subdir6)
+        self.add_device_cmd("nvme show-regs %(dev)s", whitelist='nvme.*',
+                            devices='block', subdir=_subdir7)
```

Comment:
An option that *can* lead to more concise and readable code:

    for cmd, subdir in [(smartctl_cmds, _subdir1),
                                      (nvme_logs_cmds, _subdir2),
                                      ..
                                      ("nvme show-regs %(dev)s", _subdir7)]:
        self.add_device_cmd(cmd, whitelist='nvme.*', devices='block', subdir=subdir)

It is a matter of subjective opinion what is better here, imho, and I am OK with either approach - leaving on the author to decide.

Reply:
Apologies for the delay. I have made the necessary changes and shall send the modified path by tomorrow.
Thank you.

Reply:
I have pushed changes to https://github.com/Nitn-Yewale/sos. Could you please check now. Please let me know if any action item is on me.